src/utils/tst/StringSearch.cpp

```diff
@@ -0,0 +1,60 @@
+#include "UtilTestFixture.h"
+
+class StringSearchFunctionalityTest : public UtilTestBase {
+};
+
+TEST_F(StringSearchFunctionalityTest, NegativeInvalidInput) {
```

Comment:
Add some tests for empty. Also, add cases where needle is larger than the stack

Reply:
Makes sense! I've added the unit tests.